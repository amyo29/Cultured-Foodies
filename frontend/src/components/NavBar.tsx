import React, { Component } from "react";
import {
  Container,
  Row,
  Col,
  Navbar,
  NavDropdown,
  Nav,
  Form,
  FormControl,
  Button,
} from "react-bootstrap";

import "../styles/NavBar.css"

class NavBar extends Component {
  render() {
    return (
      <div>
        <Navbar expand="lg" className="navbar-custom">
          <Navbar.Brand href="/" className="nav-link">
            <b>Cultured Foodies</b>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="mr-auto">
              <Nav.Link href="/about">About Us</Nav.Link>
              <Nav.Link href="/countries">Countries</Nav.Link>
              <Nav.Link href="/dishes">Dishes</Nav.Link>
              <Nav.Link href="/news">News</Nav.Link>
            </Nav>
            <Form inline>
              <FormControl
                type="text"
                placeholder="Search"
                className="mr-sm-2"
              />
              <Button variant="outline-success">Search</Button>
            </Form>
          </Navbar.Collapse>
        </Navbar>
      </div>
    );
  }
}

export default NavBar;
