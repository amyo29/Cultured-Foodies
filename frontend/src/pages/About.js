import React from 'react';
import {Button} from 'react-bootstrap';

function About() {
  return (
    <div >
      <header >
        about us
      </header>
    </div>
  );
}

export default About;