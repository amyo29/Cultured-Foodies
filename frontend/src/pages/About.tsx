import React from 'react';
// import {Button} from 'react-bootstrap';

function About() {
  return (
    <div>
      <h2>
        About
      </h2>
      <td>
        <p>
          <b>one</b> page
        </p>
        <ul>
          <li>description of the site, its purpose, its intended users</li>
          <li>explanation of the interesting result of integrating disparate data</li>
          <li>the group members</li>
          <li>for each member
          <ul>
              <li>name</li>
              <li>photo</li>
              <li>bio</li>
              <li>major responsibilities</li>
              <li>no. of commits</li>
              <li>no. of issues</li>
              <li>no. of unit tests</li>
            </ul>
          </li>
          <li>stats
          <ul>
              <li>total no. of commits</li>
              <li>total no. of issues</li>
              <li>total no. of unit tests</li>
            </ul>
          </li>
          <li>data
          <ul>
              <li>links to the data <b>sources</b></li>
              <li>description of <b>how</b> each was <b>scraped</b></li>
            </ul>
          </li>
          <li>tools
          <ul>
              <li>tools used</li>
              <li>describe their use</li>
              <li>special focus on optional tools that were <b>not</b> required</li>
            </ul>
          </li>
          <li>a link to the <a href="https://gitlab.com/cs373-group-11/cultured-foodies">GitLab</a> repo</li>
          <li>a link to the <a href="https://www.postman.com">Postman</a> API</li>
        </ul>
      </td>
    </div>
  );
}

export default About;
