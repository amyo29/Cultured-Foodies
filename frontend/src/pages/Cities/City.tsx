import React, { useEffect, useState } from "react";
import { Jumbotron } from "react-bootstrap";
import Image from "react-bootstrap/Image";
import { Container } from "react-bootstrap";
import { Row } from "react-bootstrap";
import { Col } from "react-bootstrap";
import { useParams } from "react-router-dom";
import useAxios from 'axios-hooks';
import axios from 'axios';
import {RestaurantInstance} from "../Restaurants/Restaurant";
import {CuisineInstance} from "../Cuisines/Cuisine";



function City() {
  useEffect(() => {
    document.title = "City";
  }, []);
  const { id } = useParams<{ id: string }>();
  const [{ data, loading, error }] = useAxios('/api/cities/id='+id) 
  const [city, setCity] = useState<CityInstance>();
  const [restaurants, setRestaurants] = useState<Array<RestaurantInstance>>();
  const [cuisines, setCuisines] = useState<Array<CuisineInstance>>();

  /* set city data */
  function matchesCityCuisines(element:any, index: number, array: any){ 
    return city?.cuisine_ids.split(", ").includes(element?.id.toString())
} 
function matchesCityRestaurants(element:any, index: number, array: any){ 
  return city?.restaurant_ids.split(", ").includes(element?.id.toString())
} 
  useEffect(() => {
    const cityObj: CityInstance = data as CityInstance;
    if (cityObj) {
      setCity(cityObj);
    }
  }, [data]);


  useEffect(() => {
    axios.get("/api/cuisines").then((value) => {
      let all_cuisines = value["data"]["cuisines"]
      let filtered_cuisines = all_cuisines.filter(matchesCityCuisines)
      console.log('cuisine', filtered_cuisines)
      setCuisines(filtered_cuisines)
    });
    axios.get("/api/restaurants").then((value) => {
      let all_restaurants = value["data"]["restaurants"]
      let filtered_restaurants = all_restaurants.filter(matchesCityRestaurants)
      setRestaurants(filtered_restaurants)
      console.log('rest', filtered_restaurants)
    });
  }, [city])
  
  //render: map long/lat or city names to render map
  let API_KEY = "AIzaSyBnpJl9h_gz0umc1sVng27AS3rNZOg7LR8";
  let cityMapURL =
    "https://www.google.com/maps/embed/v1/place?key=" +
    API_KEY +
    "&q=" +
    city?.full_name;



  return (

    <Container > 
      <Image src={city?.imagesweb} fluid />

      <h1>{city?.name}</h1> 

      <section>
        <h2>About the City</h2>
        {city?.full_name}

        <h5>State</h5>
        {city?.state}

        <h5>Population</h5>
        {city?.population}

        <h5>Location</h5>
        <h6>Latitude</h6>
        {city?.latitude}
        <h6>Longitude</h6>
        {city?.longitude}

      </section>

      <section style={{alignItems:"left", paddingLeft:"1px"}}>  
        <h5>Scores</h5>
        <h6>Business Freedom</h6>
        {city?.business_freedom}

        <h6>Cost of Living</h6>
        {city?.cost_of_living}

        <h6>Economy</h6>
        {city?.economy}

        <h6>Housing</h6>
        {city?.housing}

        <h6>Commute</h6>
        {city?.commute}

        <h6>Education</h6>
        {city?.education}

        <h6>Environmental Quality</h6>
        {city?.environmental_quality}

        <h6>Healthcare</h6>
        {city?.healthcare}

        <h6>Leisure and Culture</h6>
        {city?.leisure_culture}

        <h6>Outdoors</h6>
        {city?.outdoors}

        <h6>Safety</h6>
        {city?.safety}

        <h6>Taxation</h6>
        {city?.taxation}

        <h6>Travel and Connectivity</h6>
        {city?.travel_connectivity}

        <h6>Internet Access</h6>
        {city?.internet_access}

        <h6>Venture Capital</h6>
        {city?.venture_capital}

      </section>
      <section>
      <h5>Restauarants in {city?.name}</h5>
      {restaurants?.map((r) => (<a href={"/restaurants/" + r.id}>{r.name}<br/></a>))}
      <h5>Cuisines {city?.name}</h5>
      {cuisines?.map((c) => (<a href={"/restaurants/" + c.id}>{c.name}<br/></a>))}
      </section>


        <div className="center">
          <h4>Map Location</h4>
          <iframe src={cityMapURL} width="600" height="450" loading="lazy">
          </iframe>
        </div> 

        {/* <h5>Health</h5>
        {data["recipe"]["healthLabels"].map((healthLabel: any) => {
          return <li>{healthLabel}</li>;
        })}
      </section>

      <Image src={data["recipe"]["image"]} fluid />

      <p>{data["recipe"]["source"]}</p>

      <h5>Calories: {data["recipe"]["calories"]}</h5>
      <h5>Total energy (kcal): {data["recipe"]["totalNutrients"]["ENERC_KCAL"]["quantity"]}</h5>

      <p>
        <h5>Ingredients</h5>
        {data["recipe"]["ingredientLines"].map((ingredient: any) => {
          return <li>{ingredient}</li>;
        })}
      </p>

      <h5>Restrictions</h5>
      {data["recipe"]["cautions"].length == 0
        ? "No restrictions"
        : data["recipe"]["cautions"].map((caution: any) => {
            return <li>{caution}</li>;
          })
      }

      <h5>Country: </h5>
      <a href={"/countries/" + countryIndex}>{country["name"]}</a>

      <h5>Mentioned in news articles: </h5>
      <a href={"/news/" + newsIndex}>{article["title"]}</a>
      <br />

      <div className="center">
        <h3>Recipe video</h3>
        <iframe
          src={youTubeURL}
          width="50%"
          height="400"
          allow="fullscreen"
          title="video"
        />
      </div> */}
    </Container>
  );
}


export interface CityInstance {
  business_freedom: number;
  commute: number;
  cost_of_living: number;
  economy: number;
  education: number;
  environmental_quality: number;
  full_name: string;
  healthcare: number;
  housing: number;
  id: number;
  imagesmobile: string;
  imagesweb: string;
  internet_access: number;
  latitude: number;
  leisure_culture: number;
  longitude: number;
  name: string;
  outdoors: number;
  population: number;
  safety: number;
  state: string;
  taxation: number;
  timezone: string;
  travel_connectivity: number;
  venture_capital: number;
  cuisine_ids: string;
  restaurant_ids: string;

}


export default City;
