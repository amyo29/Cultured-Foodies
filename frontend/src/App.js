import React from 'react';
import {BrowserRouter, Switch, Route} from 'react-router-dom'
import Homepage from './pages/Homepage'
import About from './pages/About'
import './App.css';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={Homepage}/>
          <Route exact path="/about" component={About}/>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;