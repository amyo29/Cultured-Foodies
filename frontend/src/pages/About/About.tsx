import React, { useEffect, useState } from "react";
import { API_INFO, TOOLS_INFO, TEAM_INFO } from "./AboutInfo";
import { InfoCard, ProfileCard } from "../../components/Card";
import retrieveGitLabInfo, {
  TOTAL_COMMITS_INDEX,
  TOTAL_ISSUES_INDEX,
  TOTAL_TESTS_INDEX,
} from "./GitLabInfo";
import "../../styles/About.css";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import Grid, { GridSpacing } from "@material-ui/core/Grid";
import { CardDeck } from "react-bootstrap";
import { Container, Row, Col, Card, ListGroup, Navbar } from "react-bootstrap";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    paper: {
      height: 140,
      width: 100,
    },
    control: {
      padding: theme.spacing(2),
    },
  })
);


function About() {
  useEffect(() => {
    document.title = "About"
  }, [])
  const [spacing, setSpacing] = React.useState<GridSpacing>(2);
  const classes = useStyles();

  const [commitsSum, changeCommitsSum] = useState(-1);
  const [issuesSum, changeIssuesSum] = useState(-1);
  const [testsSum, changeTestsSum] = useState(-1);
  const [teamData, changeTeamData] = useState([{}]);
  const [loaded, changeLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      if (teamData === undefined || teamData.length === 1) {
        const gitLabInfo = await retrieveGitLabInfo();
        changeCommitsSum(gitLabInfo.statsInfo[TOTAL_COMMITS_INDEX]);
        changeIssuesSum(gitLabInfo.statsInfo[TOTAL_ISSUES_INDEX]);
        changeTestsSum(gitLabInfo.statsInfo[TOTAL_TESTS_INDEX]);
        changeTeamData(gitLabInfo.TEAM_INFO);
        changeLoading(true);
      }
    };
    fetchData();
  }, [teamData]);

  const teamData1 = teamData.slice(0, 3);
  const teamData2 = teamData.slice(3, 6);
  var i, j;
  var chunk = 3;
  var toolRows = [];
  for (i = 0, j = TOOLS_INFO.length; i < j; i += chunk) {
    toolRows.push(TOOLS_INFO.slice(i, i + chunk));
  }
  return (
    <div className="font-style center">
      <h1>About Us</h1>

      <p>
        Cultured Foodies is a website that encourages its users to discover new
        foods from different cultures. We want people to explore the world of
        food from our website and learn more about new cuisines and dishes.
      </p>

      <p>
        The data we've gathered highlights cuisines from various cultures and
        their known presence in news outlets.
      </p>

      <h2>Meet the Team</h2>
      <div className="row main-card">
        <div className="col-7 card-deck">
            {loaded ? (
              teamData1.map((teamMember: any) => {
                const {
                  name,
                  img,
                  role,
                  bio,
                  commits,
                  issues,
                  tests,
                } = teamMember;

                return (
                  <ProfileCard
                    name={name}
                    img={img}
                    role={role}
                    bio={bio}
                    commits={commits}
                    issues={issues}
                    tests={tests}
                  />
                );
              })
            ) : (
              <div>Loading</div>
            )}
        </div>
      </div>
      <div className="row main-card">
        <div className="col-7 card-deck">
            {loaded ? (
              teamData2.map((teamMember: any) => {
                const {
                  name,
                  img,
                  role,
                  bio,
                  commits,
                  issues,
                  tests,
                } = teamMember;

                return (
                  <ProfileCard
                    name={name}
                    img={img}
                    role={role}
                    bio={bio}
                    commits={commits}
                    issues={issues}
                    tests={tests}
                  />
                );
              })
            ) : (
              <div>Loading</div>
            )}
        </div>
      </div>

      <div>
        <h2>Stats</h2>
        <div>
          {commitsSum === -1 ? (
            <h5>Total Commits: </h5>
          ) : (
            <h5>Total Commits: {commitsSum}</h5>
          )}
        </div>
        <div>
          {commitsSum === -1 ? (
            <h5>Total Issues: </h5>
          ) : (
            <h5>Total Issues: {issuesSum}</h5>
          )}
        </div>
        <div>
          {commitsSum === -1 ? (
            <h5>Total Tests: </h5>
          ) : (
            <h5>Total Tests: {testsSum}</h5>
          )}
        </div>
      </div>

      <div>
        <h2>API's</h2>
      </div>

      <Grid container className={classes.root} spacing={2} justify="center">
        {API_INFO.map((api: any) => {
          const { title, img, description, link } = api;

          return (
            <InfoCard
              title={title}
              img={img}
              description={description}
              link={link}
            />
          );
        })}
      </Grid>

      <div>
        <h2>Tools</h2>
      </div>


       <Container>
        {toolRows.map((cols) => (
          <Row>
            {cols.map((col: any, i: any) => (
              <Col>
                <InfoCard
                  title={col.title}
                  img={col.img}
                  description={col.description}
                  link={col.link}
                />
              </Col>
            ))}
          </Row>
        ))}
      </Container>
      <div>
        <h2>Our Data</h2>
      </div>
      <Grid container className={classes.root} spacing={2} justify="center">
        <InfoCard
          title="Gitlab Repo"
          img={TOOLS_INFO[1].img}
          link="https://gitlab.com/cs373-group-11/cultured-foodies"
        />
        <InfoCard
          title="Postman API"
          img={TOOLS_INFO[2].img}
          link="https://documenter.getpostman.com/view/14740527/Tz5i9gAY"
        />
      </Grid>
    </div>
  );
}

export default About;
