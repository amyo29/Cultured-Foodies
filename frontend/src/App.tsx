import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./styles/App.css";
import SplashVertical from "./pages/Home/SplashVertical";
import About from "./pages/About/About";
import Cuisines from "./pages/Cuisines/Cuisines";
import Cuisine from "./pages/Cuisines/Cuisine";
import Cities from "./pages/Cities/Cities";
import City from "./pages/Cities/City";
import Restaurants from "./pages/Restaurants/Restaurants";
import Restaurant from "./pages/Restaurants/Restaurant";
import Search from "./pages/Search/Search";
import Visualization from "./pages/Visualizations";
import ProviderVisualizations from "./pages/ProviderVisualizations";

export default function App() {
  return (
    <div className="background full-height">
      <Router>
        <div>
          <Route exact path="/">
            <SplashVertical />
          </Route>
          {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
          <Switch>
            <Route path="/about">
              <About />
            </Route>
            <Route exact path="/cuisines">
              <Cuisines />
            </Route>
            <Route
              path="/cuisines/:id"
              render={(props) => <Cuisine id={props.match.params.id} />}
            />
            <Route exact path="/cities">
              <Cities />
            </Route>
            <Route
              path="/cities/:id"
              render={(props) => <City id={props.match.params.id} />}
            />
            <Route exact path="/restaurants">
              <Restaurants />
            </Route>
            <Route
              path="/restaurants/:id"
              render={(props) => <Restaurant id={props.match.params.id} />}
            />
            <Route path="/visualizations">
              <Visualization />
            </Route>
            <Route path="/provider-visualizations">
              <ProviderVisualizations />
            </Route>
            <Route
              path="/search/query=:q"
              render={(props) => <Search q={props.match.params.q} />}
            />
          </Switch>
        </div>
      </Router>
    </div>
  );
}
