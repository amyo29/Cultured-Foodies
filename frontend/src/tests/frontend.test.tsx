
import React from 'react';
import ReactDOM from 'react-dom';

//import { expect } from 'chai';
import * as Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, configure } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import App from "../App";
import NavBar from "../components/NavBar";  
import Home from "../pages/Home/Home";
import About from "../pages/About/About";
import Restaurants from "../pages/Restaurants/Restaurants";
import Restaurant from "../pages/Restaurants/Restaurant";
import Cities from "../pages/Cities/Cities";
import City from "../pages/Cities/City";
import Cuisines from "../pages/Cuisines/Cuisines";
import Cuisine from "../pages/Cuisines/Cuisine";

  
configure({ adapter: new Adapter() });

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe('<Navigation />', () => {
	it('renders a Navigation Bar', () => {
	  const nav = shallow(<NavBar />);

      expect(nav).toMatchSnapshot();
	  
	});
});

// it('Cities', async () => {
//     const component = shallow(<Cities/>);
//     const data = component.instance();
//     data.componentDidMount();
//     expect(component).toMatchSnapshot();	
  
//   });
  
// it('City', async () => {
//     const component = shallow(<City/>);
//     const data = component.instance();
//     await data.componentDidMount();
//     expect(component).toMatchSnapshot();
// });
  

describe("Render views", () => {
	// test for Splash and About
	test('Splash Page', () => {
		const component = shallow(<Home />);
		expect(component).toMatchSnapshot();
	});
	test('About', () => {
		const component = shallow(<About />);
		expect(component).toMatchSnapshot();
	});
	// tests for model pages
	test('Restaurants', () => {
		const component = shallow(<Restaurants />);
		expect(component).toMatchSnapshot();
	});
	test('Cities', () => {
		const component = shallow(<Cities />);
		expect(component).toMatchSnapshot();
	});
	test('Cuisines', () => {
		const component = shallow(<Cuisines />);
		expect(component).toMatchSnapshot();
	});
	// tests for instance pages
	// need to restructure frontend to make these tests pass
	//     TypeError: Cannot read property 'match' of undefined
	// this error is due to using useParams from react-router (see https://stackoverflow.com/questions/59153898/typeerror-cannot-read-property-match-of-undefined-when-using-useparams-from-r)
	
	// test('Restaurant', () => {
	// 	const component = shallow(<Restaurant/>);
	// 	expect(component).toMatchSnapshot();
	// });
	// test('City', () => {
	// 	const component = shallow(<City/>);
	// 	expect(component).toMatchSnapshot();
	// });
	// test('Cuisine', () => {
	// 	const component = shallow(<Cuisine/>);
	// 	expect(component).toMatchSnapshot();
	// });
})

/* about page test 
it('Get About Page', async () => {
	const copy = shallow(<About />);
	expect(copy).to.not.be.undefined;
	expect(copy).to.have.length(1);
	expect(copy.find("div")).to.have.length(11);
});
*/
