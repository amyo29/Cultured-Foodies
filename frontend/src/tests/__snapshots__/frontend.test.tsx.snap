// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders a Navigation Bar 1`] = `
<div>
  <Navbar
    className="navbar-custom"
    collapseOnSelect={false}
    expand="lg"
    variant="light"
  >
    <NavbarBrand
      className="nav-link"
      href="/"
    >
      <b>
        Cultured Foodies
      </b>
    </NavbarBrand>
    <NavbarToggle
      aria-controls="basic-navbar-nav"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="basic-navbar-nav"
    >
      <Nav
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          href="/about"
        >
          About Us
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          href="/cuisines"
        >
          Cuisines
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          href="/cities"
        >
          Cities
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          href="/restaurants"
        >
          Restaurants
        </NavLink>
      </Nav>
      <Form
        inline={true}
      >
        <FormControl
          className="mr-sm-2"
          placeholder="Search"
          type="text"
        />
        <Button
          active={false}
          disabled={false}
          variant="outline-success"
        >
          Search
        </Button>
      </Form>
    </NavbarCollapse>
  </Navbar>
</div>
`;

exports[`Render views About Page 1`] = `
<div
  className="font-style center"
>
  <h1>
    About Us
  </h1>
  <p>
    Hong Kong's dim sum? Italy's lasagna? Or UK's fish 'n' chips? Join us on a journey of culinary and cultural discovery that stretches through the ages and across the seas. With information on delectable cuisines from around the world and where you can experience them in your city, as well as highlights on worldwide countries and U.S cities, Cultured Foodies makes it easy for you to feast to your mind and stomach's content.
  </p>
  <h2>
    Meet the Team
  </h2>
  <Container
    fluid={false}
  >
    <div
      className="row mb-4"
    >
      <div>
        Loading
      </div>
    </div>
  </Container>
  <Container
    fluid={false}
  >
    <div
      className="row mb-4"
    >
      <div>
        Loading
      </div>
    </div>
  </Container>
  <div>
    <h2>
      Stats
    </h2>
    <div>
      <h5>
        Total Commits: 
      </h5>
    </div>
    <div>
      <h5>
        Total Issues: 
      </h5>
    </div>
    <div>
      <h5>
        Total Tests: 
      </h5>
    </div>
  </div>
  <div>
    <h2>
      APIs
    </h2>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="row mb-4"
    >
      <InfoCard
        description="Used to retrieve country information"
        img="RestCountriesLogo.png"
        link="https://restcountries.eu/"
        title="REST Countries"
      />
      <InfoCard
        description="Used to find restaurant data."
        img="zomato.png"
        link="https://rapidapi.com/blog/how-to-use-zomato-api/"
        title="Zomato API"
      />
      <InfoCard
        description="Used to retrieve cuisine and dishes data."
        img="WorldFood.png"
        link="https://worldfood.guide/api/"
        title="WorldFood API"
      />
    </div>
    <div
      className="row mb-4"
    >
      <InfoCard
        description="Used to show the map location of a country"
        img="GoogleMapsLogo.jpeg"
        link="https://developers.google.com/maps/documentation/embed/get-started"
        title="Google Maps Embed API"
      />
      <InfoCard
        description="Used to retrieve images of restaurants."
        img="GooglePlaceLogo.png"
        link="https://developers.google.com/maps/documentation/places/web-service/photos"
        title="Google Places Photos API"
      />
      <InfoCard
        description="Used to retrieve city data."
        img="teleportAPI.png"
        link="https://developers.teleport.org/api/"
        title="Teleport API"
      />
    </div>
  </Container>
  <div>
    <h2>
      Tools
    </h2>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="row mb-4"
    >
      <InfoCard
        description="Cloud hosting platform"
        img="AWSLogo.png"
        link="https://aws.amazon.com/"
        title="AWS"
      />
      <InfoCard
        description="GitLab Repository and CI/CD platform"
        img="GitLabLogo.png"
        link="https://gitlab.com/"
        title="GitLab"
      />
      <InfoCard
        description="API and streamline collaboration platform"
        img="PostmanLogo.png"
        link="https://www.postman.com/"
        title="Postman"
      />
    </div>
    <div
      className="row mb-4"
    >
      <InfoCard
        description="A JavaScript library for building user interfaces"
        img="ReactLogo.png"
        link="https://reactjs.org/"
        title="React"
      />
      <InfoCard
        description="A free and open-source CSS framework directed at responsive, mobile-first front-end web development"
        img="BootstrapLogo.jpg"
        link="https://getbootstrap.com/"
        title="BootStrap"
      />
      <InfoCard
        description="Domain name registrar"
        img="Namecheap.png"
        link="https://www.namecheap.com/"
        title="NameCheap"
      />
    </div>
    <div
      className="row mb-4"
    >
      <InfoCard
        description="Open-source relational database management system."
        img="PostgreSQL.png"
        link="https://postgresql.org"
        title="PostgreSQL"
      />
    </div>
  </Container>
  <div>
    <h2>
      Our Data
    </h2>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="row justify-content-center"
    >
      <InfoCard
        img="GitLabLogo.png"
        link="https://gitlab.com/cs373-group-11/cultured-foodies"
        title="Gitlab Repo"
      />
      <InfoCard
        img="PostmanLogo.png"
        link="https://documenter.getpostman.com/view/14712963/Tz5jczL8"
        title="Postman API"
      />
    </div>
  </Container>
</div>
`;

exports[`Render views Cities model page 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "inline-block",
      "justifyContent": "horizontal",
    }
  }
>
  <h1
    className="text-align center"
  >
    Cities
  </h1>
  <Container
    fluid={false}
  />
  <div
    className="row pagination"
  >
    <WithStyles(ForwardRef(Pagination))
      count={0}
      onChange={[Function]}
      page={1}
    />
    1
     - 
    0
     of 
    0
  </div>
</div>
`;

exports[`Render views City instance page 1`] = `
<Container
  fluid={false}
>
  <Image
    fluid={true}
    rounded={false}
    roundedCircle={false}
    thumbnail={false}
  />
  <h1 />
  <section>
    <h2>
      About the City
    </h2>
    <h5>
      Summary
    </h5>
    <h5>
      State
    </h5>
    <h5>
      Population
    </h5>
    <h5>
      Location
    </h5>
    <h6>
      Latitude
    </h6>
    <h6>
      Longitude
    </h6>
  </section>
  <section
    style={
      Object {
        "alignItems": "left",
        "paddingLeft": "1px",
      }
    }
  >
    <h5>
      Scores
    </h5>
    <h6>
      Business Freedom
    </h6>
    <h6>
      Cost of Living
    </h6>
    <h6>
      Economy
    </h6>
    <h6>
      Housing
    </h6>
    <h6>
      Commute
    </h6>
    <h6>
      Education
    </h6>
    <h6>
      Environmental Quality
    </h6>
    <h6>
      Healthcare
    </h6>
    <h6>
      Leisure and Culture
    </h6>
    <h6>
      Outdoors
    </h6>
    <h6>
      Safety
    </h6>
    <h6>
      Taxation
    </h6>
    <h6>
      Travel and Connectivity
    </h6>
    <h6>
      Internet Access
    </h6>
    <h6>
      Venture Capital
    </h6>
  </section>
  <section>
    <h5>
      Restaurants in 
    </h5>
    <p>
      loading
    </p>
    <h5>
      Cuisines of 
    </h5>
  </section>
  <div
    className="center"
  >
    <h4>
      Map Location
    </h4>
    <iframe
      height="450"
      loading="lazy"
      src="https://www.google.com/maps/embed/v1/place?key=AIzaSyBnpJl9h_gz0umc1sVng27AS3rNZOg7LR8&q=undefined"
      width="600"
    />
  </div>
</Container>
`;

exports[`Render views Cuisine instance page 1`] = `
<Container
  fluid={true}
>
  <Component />
  <section>
    <h5>
      Restaurants with 
       food
    </h5>
    <p>
      loading
    </p>
    <h5>
      Cities with 
       food
    </h5>
    <p>
      loading
    </p>
  </section>
  <p />
  <p />
  <p />
  <p />
</Container>
`;

exports[`Render views Cuisines model page 1`] = `
<div>
  <h1
    className="text-align center"
  >
    Cuisines
  </h1>
  <Container
    fluid={false}
  />
  <div
    className="row pagination"
  >
    <WithStyles(ForwardRef(Pagination))
      count={0}
      onChange={[Function]}
      page={1}
    />
    1
     - 
    0
     of 
    0
  </div>
</div>
`;

exports[`Render views Restaurant instance page 1`] = `
<div>
  <body
    className="Instance center-div"
  >
    <h2
      className="Instance-header"
    />
    <div
      align-items="center"
      className="info"
    >
      <div
        className="row"
      >
        <img
          alt=""
          className="col instance-photo"
        />
        <Card
          body={false}
          className="col instance-text"
          style={
            Object {
              "width": "55rem",
            }
          }
        >
          <CardBody>
            <CardTitle />
            <CardText>
              Address: 
            </CardText>
            <CardText>
              Average Rating: 
            </CardText>
            <CardText>
              Highlights: 
            </CardText>
            <CardText>
              Price Range: 
            </CardText>
            <CardText>
              Timings: 
            </CardText>
            <CardText>
              Menu:
              <a>
                <br />
              </a>
            </CardText>
            <CardText>
              Cities which have this restaurant:
              <a
                href="/cities/undefined"
              >
                 
                <br />
              </a>
            </CardText>
            Cuisines for this restaurant: 
            <br />
            <p>
              loading
            </p>
            <CardText />
          </CardBody>
        </Card>
      </div>
    </div>
  </body>
  <div
    className="center"
  >
    <h4>
      Map Location
    </h4>
    <iframe
      height="450"
      loading="lazy"
      src="https://www.google.com/maps/embed/v1/place?key=AIzaSyBnpJl9h_gz0umc1sVng27AS3rNZOg7LR8&q=undefined"
      width="600"
    />
  </div>
</div>
`;

exports[`Render views Restaurants model page 1`] = `
<div>
  <h1
    className="text-align center"
  >
    Restaurants
  </h1>
  <ForwardRef
    className="table"
    responsive={true}
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Name
        </th>
        <th>
          City
        </th>
        <th>
          Cuisines
        </th>
        <th>
          Rating
        </th>
        <th>
          Price Range
        </th>
        <th>
          Average Cost for Two
        </th>
      </tr>
    </thead>
    <tbody />
  </ForwardRef>
  <div
    className="row pagination"
  >
    <WithStyles(ForwardRef(Pagination))
      count={0}
      onChange={[Function]}
      page={1}
    />
    1
     - 
    0
     of 
    0
  </div>
</div>
`;

exports[`Render views Splash Page 1`] = `
<div>
  <body
    className="Home"
    data-background="homephoto.jpg"
    id="mybody"
  >
    <header
      className="Home-header"
    >
      <div
        className="Title-summary"
      >
        Welcome fellow Foodie!
        <div
          className="summary"
        >
          Whether you are looking for cultural dishes
          <br />
          around the world, country information, dishes 
          <br />
          or global news, Cultured Foodies 
          <br />
          can help you find what you're looking for!
          <br />
        </div>
      </div>
    </header>
  </body>
</div>
`;
